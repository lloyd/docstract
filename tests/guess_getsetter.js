// __defineGetter__ / __defineSetter__ suggests a property, name is first
// arg to function.

  /**
   * True iff the stream is closed.
   */
stream.__defineGetter__("closed", function stream_closed() {
    return !self.opened;
});

  /**
   * True iff the stream is open.
   */
stream.__defineSetter__('opened', function(x) {
    self.opened = x;
});
