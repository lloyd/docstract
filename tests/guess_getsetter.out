{
  "filename": "tests/guess_getsetter.js", 
  "module": "guess_getsetter", 
  "properties": [
    {
      "desc": "True iff the stream is closed.", 
      "name": "closed", 
      "source_lines": [
        4, 
        6
      ]
    }, 
    {
      "desc": "True iff the stream is open.", 
      "name": "opened", 
      "source_lines": [
        11, 
        13
      ]
    }
  ]
}
